src/app/api/booking/route.ts

```diff
@@ -0,0 +1,117 @@
+import { NextResponse } from "next/server";
+import { and, asc, eq, gt, lt, sql } from "drizzle-orm";
+
+import { db } from "@/lib/db";
+import { bookings, gameSessions, users } from "@/lib/db/schema";
+import { getCurrentUser } from "@/lib/session";
+
+/**
+ * Gets game sessions currently available for booking
+ */
+
+interface Session {
+  gamesessionId: number;
+  playLevel: string;
+}
+
+export async function POST(request: Request) {
+  try {
+    const currentUser = await getCurrentUser();
+    if (currentUser === null)
+      return new Response("unauthorized user", { status: 401 });
+
+    const json = await request.json();
+    const num_of_sessions = json.length;
```

Comment:
consider validating the body through a zod schema

Reply:
done